src/service/plugins/sftp.js

```diff
@@ -317,6 +318,54 @@ var Plugin = GObject.registerClass({
         }
     }
 
+    /**
+     * Create a symbolic link referring to the device by name
+     */
+    _addSymlink(mount) {
+        let by_name_dir = Gio.File.new_for_path(
+            gsconnect.runtimedir + '/by-name/'
+        );
+        try {
+            by_name_dir.make_directory_with_parents(null);
+        } catch (e) {
+            if ( ! e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.EXISTS) ) {
+                throw e;
+            }
+        }
+
+        // Replace path separator with a Unicode lookalike:
+        let safe_device_name = this.device.name.replace('/', '∕');
+        if ( safe_device_name == '.' ) {
+            safe_device_name = '·';
+        }
+        else if ( safe_device_name == '..' ) {
+            safe_device_name = '··';
+        }
+
+        let link_target = mount.get_root().get_path();
+        let link = Gio.File.new_for_path(
+            by_name_dir.get_path() + '/' + safe_device_name
+        );
+
+        // Check for and remove any existing stale link:
+        try {
+            let link_stat = link.query_info(
```

Comment:
When `_addSymlink()` is made async, you can wrap this and `g_file_delete()` in small promises:

```js
let link_stat = await new Promise((resolve, reject) => {
    link.query_info_async(
        "standard::symlink-target",
        Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS,
        GLib.PRIORITY_DEFAULT,
        null,
        (link, res) => {
        try {
            resolve(link.query_info_finish (res));
        } catch (e) {
            reject(e);
        }
    );
});
```

In the case of `g_file_delete_async()` below, you can just use `debug()` on a NOT_FOUND error and call `resolve()` from the catch.